bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/BDBMessageStoreTest.java

```diff
@@ -218,4 +228,25 @@ protected boolean flowToDiskSupported()
         return true;
     }
 
+    @Test
+    public void testGetNextMessgaeIdWithLockException(){
+        BDBMessageStore bdbStore = (BDBMessageStore) createMessageStore();
```

Comment:
The test mixes different mocking approaches: spy, mockito moc and mocked messages for AMQP 0-8.
IMHO, we can get rid of spy by using mockito mocks for EnvironmentFacadeFactory, EnvironmentFacade and message
```
EnvironmentFacade eff = mock(EnvironmentFacadeFactory.class);
EnvironmentFacade ef = mock(EnvironmentFacade.class);
doThrow(LockTimeoutException.class).when(ef).openSequence(any(),any(),any());
when(eff.createEnvironmentFacade(any())).thenReturn(ef);

BDBMessageStore store = new BDBMessageStore (eff);
store.openMessageStore();

 try
        {
            store.addMessage(new InternalMessageMetaData(false, null, 0))
            fails("exception is expected");
        }
        catch(ConnectionScopedRuntimeException e)
        {
            assertEquals("Unexpected exception on BDB sequence",e.getMessage());
        }

```

I think the above reduces dependency to amqp 0-8 protocol and gets rid of spy

Reply:
Changed it!